src/cockpit/config.py

```diff
@@ -24,11 +25,29 @@
 
 logger = logging.getLogger(__name__)
 
-ETC_COCKPIT = Path('/etc/cockpit')
 XDG_CONFIG_HOME = Path(os.getenv('XDG_CONFIG_HOME') or os.path.expanduser('~/.config'))
 DOT_CONFIG_COCKPIT = XDG_CONFIG_HOME / 'cockpit'
 
 
+@functools.lru_cache()
+def lookup_config(filename: str) -> Path:
```

Comment:
I meant that this should look like

```python
@functools.lru_cache()
def lookup_config() -> Path:
```

and get used like (unrealistic strawman)

```
   conf_text = (lookup_config() / 'cockpit.conf').read_text()
```

Reply:
Okay.  I misread the function, sorry.  I thought it was finding the first *directory* that existed, but it's rather doing it per-file.  That makes the caching thing less appropriate, indeed, and kinda mitigates the entire advantage of the Pathlib approach.... :(

Reply:
As discussed in chat, I dropped the lru_cache again.